cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -549,6 +549,20 @@ hash_value_type CUDA_DEVICE_CALLABLE MurmurHash3_32<double>::operator()(double c
   return this->compute_floating_point(key);
 }
 
+template <>
+hash_value_type CUDA_DEVICE_CALLABLE
+MurmurHash3_32<numeric::decimal32>::operator()(numeric::decimal32 const& key) const
+{
+  return this->compute(key.value());
+}
+
+template <>
+hash_value_type CUDA_DEVICE_CALLABLE
+MurmurHash3_32<numeric::decimal64>::operator()(numeric::decimal64 const& key) const
```

Comment:
Template function specialization! I wonder if we can put them into a source file (existing `hashing.cu`?) and also explicitly instantiate them.

Reply:
This approach will not work with our device code because we do not compile with the `rdc` flag (relocatable device code) for performance reasons.